test/hotspot/jtreg/gc/z/TestRegistersPushPopAtZGCLoadBarrierStub.java

```diff
@@ -0,0 +1,363 @@
+                test_three_floats_push_pop_at_load_barrier(f_outer, RANDOM.nextFloat(), RANDOM.nextFloat(), RANDOM.nextFloat());
+            }
+        }
+
+        static void test_one_double() {
+            for (int i = 0; i < ITERATIONS; i++) {
+                test_one_double_push_pop_at_load_barrier(d_outer, RANDOM.nextDouble());
+            }
+        }
+
+        static void test_two_doubles() {
+            for (int i = 0; i < ITERATIONS; i++) {
+                test_two_doubles_push_pop_at_load_barrier(d_outer, RANDOM.nextDouble(), RANDOM.nextDouble());
+            }
+        }
+
+        static void test_three_doubles() {
+            for (int i = 0; i < ITERATIONS; i++) {
+                test_three_doubles_push_pop_at_load_barrier(d_outer, RANDOM.nextDouble(), RANDOM.nextDouble(), RANDOM.nextDouble());
+            }
+        }
+
+        static void test_one_vector_128() {
+            for (int i = 0; i < ITERATIONS; i++) {
+                test_one_vector_128_push_pop_at_load_barrier(f_array, f_outer);
+            }
+        }
+
+        static void test_two_vectors_128() {
+            for (int i = 0; i < ITERATIONS; i++) {
+                test_two_vectors_128_push_pop_at_load_barrier(f_array, f_outer);
+            }
+        }
+
+        static void test_three_vectors_128() {
+            for (int i = 0; i < ITERATIONS; i++) {
+                test_three_vectors_128_push_pop_at_load_barrier(f_array, f_outer);
+            }
+        }
+
+        static void test_vector_max() {
+            for (int i = 0; i < ITERATIONS; i++) {
+                test_vector_max_push_pop_at_load_barrier(f_array, f_outer);
+            }
+        }
+
+        static void test_float_and_vector() {
+            for (int i = 0; i < ITERATIONS; i++) {
+                test_float_and_vector_push_pop_at_load_barrier(f_array, f_outer, RANDOM.nextFloat());
+            }
+        }
+
+        public static void main(String args[]) {
+            Runnable r = TESTS.get(args[0]);
+            r.run();
+        }
+    }
+
+    static boolean containOnlyOneOccuranceOfKeyword(String text, String keyword) {
+        int firstIndex = text.indexOf(keyword);
+        int lastIndex = text.lastIndexOf(keyword);
+        return firstIndex != -1 && firstIndex == lastIndex;
+    }
+
+    // Check that registers are pushed and poped with correct register type and number
+    static void checkPushPopRegNumberAndType(String stdout, String keyword, String expected_freg_type,
+                                             int expected_number_of_push_pop_at_load_barrier_fregs) throws Exception {
+        String keyString = keyword + expected_number_of_push_pop_at_load_barrier_fregs + " " + expected_freg_type + " registers";
+        if (!containOnlyOneOccuranceOfKeyword(stdout, keyString)) {
+            throw new RuntimeException("Stdout is expected to contain only one occurance of keyString: " + "'" + keyString + "'");
```

Comment:
In the event of failure, would it be possible to print the erroneous output? The output from the subprocesses, being directly piped in, doesn't lend itself to easy debugging. At first I thought there might be an option that could alter OutputAnalyzers output, but sadly not.

Reply:
> In the event of failure, would it be possible to print the erroneous output? The output from the subprocesses, being directly piped in, doesn't lend itself to easy debugging. At first I thought there might be an option that could alter OutputAnalyzers output, but sadly not.

Done. Thanks for your comments.